### PR TITLE
add wechaty document

### PR DESCRIPTION
**Pull request by lijiarui**
#73 #252

### CODE REVIEW

**Comment by zixia** [comment]
Thanks! Please get at least three votes(approved) from our contributor team first, because the document needs to be improved heavily and continuously.

File: src/room.ts [hunk 1]
```diff
@@ -611,7 +893,18 @@ export class Room extends EventEmitter implements Sayable {
   }
 
   /**
-   * @todo document me
+   * Find room by topic, return all the match room
```
Comment (zixia) [inline src/room.ts]: all the matched room

File: src/room.ts [hunk 2]
```diff
@@ -669,7 +962,11 @@ export class Room extends EventEmitter implements Sayable {
   }
 
   /**
-   * @todo document me
+   * Find the room by roomId
+   *
+   * @static
```
Comment (zixia) [inline src/room.ts]: add `@private` to this function

File: src/wechaty.ts [hunk 3]
```diff
@@ -204,7 +204,16 @@ export class Wechaty extends EventEmitter implements Sayable {
   }
 
   /**
-   * @todo document me
+   * Initialize the bot, return Promise.
+   *
+   * @returns {Promise<void>}
+   * @example
+   * ```ts
+   * wechaty.init()
```
Comment (zixia) [inline src/wechaty.ts]: Please keep the variable name to be consistent with all the examples.
```ts
await bot.init()
// do other staff after bot inited
```

File: src/wechaty.ts [hunk 4]
```diff
@@ -241,40 +250,146 @@ export class Wechaty extends EventEmitter implements Sayable {
...
+   *           }
+   * 	       } else {        // 2. confirm friend ship
+   *          console.log(`new friendship confirmed with ${contact.name()}`)
+   *         }
+   *  })
+   * ```
    */
   public on(event: 'friend'     , listener: (this: Wechaty, friend: Contact, request?: FriendRequest) => void): this
   /**
    * @todo document me
```
Comment (zixia) [inline src/wechaty.ts]: Please add 'heartbeat' document if you could.

File: src/wechaty.ts [hunk 5]
```diff
@@ -241,40 +250,146 @@ export class Wechaty extends EventEmitter implements Sayable {
...
+   * wechaty.on('message', (message: Message) => {
+   *   console.log(`message ${message} received`)
+   * })
+   * ```
    */
   public on(event: 'message'    , listener: (this: Wechaty, message: Message) => void): this
+
   /**
-   * @todo document me
+   * Emit when anyone join the room
```
Comment (zixia) [inline src/wechaty.ts]: join the room -> join any room

File: src/wechaty.ts [hunk 6]
```diff
@@ -241,40 +250,146 @@ export class Wechaty extends EventEmitter implements Sayable {
...
+   * })
+   * ```
    */
   public on(event: 'room-topic' , listener: (this: Wechaty, room: Room, topic: string, oldTopic: string, changer: Contact) => void): this
+
   /**
-   * @todo document me
+   * A scan event will be emitted when the bot needs to show you a QR Code for scanning.
+   * 1. URL: {String} the QR code image URL
+   * 2.code: {Number} the scan status code. some known status of the code list here is:
```
Comment (zixia) [inline src/wechaty.ts]: 2. code

File: src/wechaty.ts [hunk 7]
```diff
@@ -360,7 +475,13 @@ export class Wechaty extends EventEmitter implements Sayable {
   }
 
   /**
-   * @todo document me
+   * Quit the bot
+   *
+   * @returns {Promise<void>}
+   * @example
+   * ```ts
+   * Wechaty.quit()
```
Comment (zixia) [inline src/wechaty.ts]: await bot.quit()

File: src/wechaty.ts [hunk 8]
```diff
@@ -393,7 +514,13 @@ export class Wechaty extends EventEmitter implements Sayable {
   }
 
   /**
-   * @todo document me
+   * Logout the bot
+   *
+   * @returns {Promise<void>}
+   * @example
+   * ```ts
+   * Wechaty.logout()
```
Comment (zixia) [inline src/wechaty.ts]: await bot.logout()

File: src/wechaty.ts [hunk 9]
```diff
@@ -409,8 +536,14 @@ export class Wechaty extends EventEmitter implements Sayable {
...
 
   /**
-   * get current user
-   * @returns {Contact} current logined user
+   * Get current user
+   *
+   * @returns {Contact}
+   * @example
+   * ```ts
+   * const contact = Wechaty.self()
```
Comment (zixia) [inline src/wechaty.ts]: Should not use `Wechaty` because it's the class name.

should instantiate an instance for Wechaty, like wechaty, and use `bot` in the example code.

```ts
const bot = Wechaty.instance()
```

File: src/contact.ts [hunk 10]
```diff
@@ -195,7 +195,8 @@ export class Contact implements Sayable {
   /**
    * Check if contact is stranger
    *
-   * @returns {boolean | null} True for not friend of the bot, False for friend of the bot, null for cannot get the info.
+   * @returns {boolean | null}
+   * True for not friend of the bot, False for friend of the bot, null for cannot get the info.
```
Comment (zixia) [inline src/contact.ts]: null for unknown.

File: src/contact.ts [hunk 11]
```diff
@@ -210,7 +211,8 @@ export class Contact implements Sayable {
   /**
    * Check if it's a offical account
    *
-   * @returns {boolean|null} True for official account, Flase for contact is not a official account
+   * @returns {boolean|null}
+   * True for official account, Flase for contact is not a official account
```
Comment (zixia) [inline src/contact.ts]: null for unknown.

File: src/contact.ts [hunk 12]
```diff
@@ -262,7 +264,8 @@ export class Contact implements Sayable {
   /**
    * Check if the contact is star contact.
    *
-   * @returns {boolean} True for star friend, False for no star friend, null for cannot get the info.
+   * @returns {boolean}
+   * True for star friend, False for no star friend, null for cannot get the info.
```
Comment (zixia) [inline src/contact.ts]: where's `null`?

File: src/message.ts [hunk 13]
```diff
@@ -319,11 +329,28 @@ export class Message implements Sayable {
   }
 
   /**
-   * @todo document me
+   *
+   * Set a sender to the message
```
Comment (zixia) [inline src/message.ts]: add `@private` to this method

File: src/message.ts [hunk 14]
```diff
@@ -613,9 +721,44 @@ export class Message implements Sayable {
...
+   *
+   * @param {string} text
+   * @param {(Contact | Contact[])} [replyTo]
+   * @returns {Promise<any>}
+   * @example
+   * ```ts
+   * const bot = Wechaty.instance()
+   * bot
+   * .on('message', async m => {
+   *   if (/^(ding|ping|bing|code)$/i.test(m.content())) {
```
Comment (zixia) [inline src/message.ts]: This `if` block is not related the example. 

Should consider deleting those codes here.

File: src/message.ts [hunk 15]
```diff
@@ -613,9 +721,44 @@ export class Message implements Sayable {
...
+   *
+   * @param {MediaMessage} mediaMessage
+   * @param {(Contact | Contact[])} [replyTo]
+   * @returns {Promise<any>}
+   * @example
+   * ```ts
+   * const bot = Wechaty.instance()
+   * bot
+   * .on('message', async m => {
+   *   if (/^(ding|ping|bing|code)$/i.test(m.content())) {
```
Comment (zixia) [inline src/message.ts]: The same as above: `if` is no need to be so complicated at here. If want to keep it, just a `ding` will be enough.

File: src/message.ts [hunk 16]
```diff
@@ -773,7 +916,19 @@ export class MediaMessage extends Message {
   }
 
   /**
-   * @todo document me
+   * Get the MediaMessage file extension, etc: `jpg`, `gif`, `pdf`, `word` ..
+   *
+   * @returns {string}
+   * @example
+   * ```ts
```
Comment (zixia) [inline src/message.ts]:
```ts
.on('message', async function (m) {
  if (m instanceof MediaMessage) {
    console.log('media message file name extention is: ' + m.ext())
  }
})
```

File: src/message.ts [hunk 17]
```diff
@@ -810,7 +965,19 @@ export class MediaMessage extends Message {
   }
 
   /**
-   * @todo document me
+   * Get the MediaMessage filename, etc: `how to build a chatbot.pdf`..
+   *
+   * @returns {string}
+   * @example
+   * ```ts
```
Comment (zixia) [inline src/message.ts]:
```ts
.on('message', async function (m) {
  if (m instanceof MediaMessage) {
    console.log('media message file name is: ' + m.filename())
  }
})
```

File: src/room.ts [hunk 18]
```diff
@@ -171,23 +173,89 @@ export class Room extends EventEmitter implements Sayable {
   }
 
   /**
-   * @todo document me
+   * Get room leave event, emitted when bot remove someone from the room or someone remove the bot from the room.
+   * If both personA and personB are not the bot itselt, the event cannot be fired if personA remove personB from the room
+   * This is Sayable for all listeners.
```
Comment (zixia) [inline src/room.ts]: `this` is Sayable for all listeners.

File: src/room.ts [hunk 19]
```diff
@@ -171,23 +173,89 @@ export class Room extends EventEmitter implements Sayable {
...
+   * Which means there will be a this.say() the method inside listener call,
```
Comment (zixia) [inline src/room.ts]: `this.say()`

File: src/room.ts [hunk 20]
```diff
@@ -171,23 +173,89 @@ export class Room extends EventEmitter implements Sayable {
...
+   *    })
+   *  }
+   * ```
    */
   public on(event: 'leave', listener: (this: Room, leaver: Contact) => void): this
+
   /**
-   * @todo document me
+   * Get room join event, emitted when someone join the room
+   * This is Sayable for all listeners.
```
Comment (zixia) [inline src/room.ts]: Need not to document the `this` is sayable so many times.

One time is enough.

File: src/room.ts [hunk 21]
```diff
@@ -171,23 +173,89 @@ export class Room extends EventEmitter implements Sayable {
...
+   *    room.on('topic', (room: Room, topic: string, oldTopic: string, changer: Contact) => {
+   *      console.log(`Room ${room.topic()} topic changed from ${oldTopic} to ${topic} by ${changer.name()}`)
+   *    })
+   *  }
    */
   public on(event: 'topic', listener: (this: Room, topic: string, oldTopic: string, changer: Contact) => void): this
+
   /**
-   * @todo document me
+   * Get EVENT_PARAM_ERROR event
```
Comment (zixia) [inline src/room.ts]: The `EVENT_PARAM_ERROR` is just a guard for IDE Intelligence.

Need not be documented, and should be set to `@private`

File: src/room.ts [hunk 22]
```diff
@@ -171,23 +173,89 @@ export class Room extends EventEmitter implements Sayable {
...
+   *
+   * @param {'EVENT_PARAM_ERROR'} event
+   * @param {() => void} listener
+   * @returns {this}
    */
   public on(event: 'EVENT_PARAM_ERROR', listener: () => void): this
+
   /**
-   * @todo document me
+   * Get RoomEventName event
```
Comment (zixia) [inline src/room.ts]: This method is the implementation of all the event listeners above.

No need to be documented too.

File: src/room.ts [hunk 23]
```diff
@@ -197,11 +265,49 @@ export class Room extends EventEmitter implements Sayable {
...
 
   /**
-   * @todo document me
+   * Send media file inside Room
+   *
+   * @param {MediaMessage} mediaMessage
+   * @example
+   * ```ts
+   * const room = await Room.find({name: 'wechaty'}) // change 'wechaty' to any of your room in wechat
+   * await room.say('/test.jpg') // put the filePath you want to send here
```
Comment (zixia) [inline src/room.ts]: `await room.say(new MediaMessage('/test.jpg'))`

File: src/room.ts (hunk 20 above)
Comment (lijiarui) [inline src/room.ts]: No, I think it should document on each function because not all of the developers will read the whole doc.

Comment (zixia) [inline src/room.ts]: So do you mean that if something is important, then we should put it everywhere?

I have to disagree with you.

**Comment by lijiarui** [comment]
here is the auto generate doc: https://github.com/lijiarui/wechaty/blob/new-doc/docs/index.md

not sure why it has double wechaty intro....

File: src/message.ts [hunk 24]
```diff
@@ -180,6 +185,11 @@ export enum AppMsgType {
   READER_TYPE              = 100001,
 }
 
+/**
+ *
+ * Enum for MsgType values.
+ * @enum {number}
+ */
```
Comment (kis87988) [inline src/message.ts]: we should include message type in to documentation

**Comment by kis87988** [comment]
maybe we should add all return promise type with notice "await"

**Comment by zixia** [comment]
I saw lots of `ISC License` in the generated document, it's not right.

The license of Wechaty project is: **Apache-2.0**

There might have some comments say "_ISC_ license", should be replaced.

**Comment by kis87988** [comment]
ISC License should be changed in whchaty.ts @line#68

**Comment by zixia** [comment]
Thanks, @kis87988, I had fixed that by removing the LICENSE info.

File: src/message.ts [hunk 25]
```diff
@@ -346,11 +407,28 @@ export class Message implements Sayable {
...
   // public to(contact?: Contact|Room|string): Contact|Room|void {
+
   /**
-   * @todo document me
+   * Set the destination as Contact for the message
+   *
+   * @param {Contact} contact
+   * @returns {void}
    */
   public to(contact: Contact): void
```
Comment (zixia) [inline src/message.ts]: This method should be marked as `@private` because normally use should not use Wechaty API to set the destination of the message by call `to()`. 

We use `Sayable`s to send message to contact, like `contact.say()` and `room.say()`

File: src/message.ts [hunk 26]
```diff
@@ -346,11 +407,28 @@ export class Message implements Sayable {
...
+
+  /**
+   * Set the destination as Contact by 'weixinID', eg: 'filehelper', for the message
+   *
+   * @param {string} id
```
Comment (zixia) [inline src/message.ts]: should be marked as `@private`

Comment (zixia) [inline src/message.ts]: should be marked as `@private`

File: src/message.ts [hunk 27]
```diff
@@ -374,10 +452,27 @@ export class Message implements Sayable {
   }
 
   /**
-   * @todo document me
+   * Set the room for a message
+   *
+   * @param {Room} room
+   * @returns {void}
    */
   public room(room: Room): void
```
Comment (zixia) [inline src/message.ts]: Should be marked as `@private`

File: src/message.ts [hunk 28]
```diff
@@ -374,10 +452,27 @@ export class Message implements Sayable {
...
+
+  /**
+   * Set the room for a message
+   *
+   * @param {string} id
+   * @returns {void}
+   */
   public room(id: string): void
```
Comment (zixia) [inline src/message.ts]: `@private`

File: src/message.ts [hunk 29]
```diff
@@ -438,16 +567,29 @@ export class Message implements Sayable {
...
    */
   public typeEx()  { return MsgType[this.obj.type] }
+
   /**
-   * @todo document me
+   * Get the serial number from the message.
+   *
+   * @returns {number}
    */
   public count()   { return this._counter }
```
Comment (zixia) [inline src/message.ts]: `@private`

File: src/message.ts [hunk 30]
```diff
@@ -584,19 +729,32 @@ export class Message implements Sayable {
...
+  /**
+   * @private
+   */
   public dumpRaw() {
     console.error('======= dump raw message =======')
     Object.keys(this.rawObj).forEach(k => console.error(`${k}: ${this.rawObj && this.rawObj[k]}`))
   }
 
+  /**
+   * @private
```
Comment (zixia) [inline src/message.ts]: add a `@todo` to mark this method is to be implemented.

File: src/message.ts [hunk 31]
```diff
@@ -584,19 +729,32 @@ export class Message implements Sayable {
...
+   */
   public static async find(query) {
     return Promise.resolve(new Message(<MsgRawObj>{MsgId: '-1'}))
   }
 
+  /**
+   * @private
+   */
   public static async findAll(query) {
```
Comment (zixia) [inline src/message.ts]: add a `@todo` to mark this method is to be implemented.

File: src/message.ts [hunk 32]
```diff
@@ -667,12 +858,34 @@ export class Message implements Sayable {
 // Message.initType()
 
 export class MediaMessage extends Message {
+  /**
+   * @private
```
Comment (zixia) [inline src/message.ts]: I think if the class property is already `private`, then we will not need to use `@private` in jsdoc again?

File: src/message.ts (hunk 17 above)
Comment (zixia) [inline src/message.ts]: Could you find a button with `resolve` text on it, near this review message?

Please help me to find it, because @binsee also need that button... thanks.

File: src/message.ts (hunk 16 above)
Comment (zixia) [inline src/message.ts]: please mark the review as `resolved` by GitHub button.

File: src/room.ts [hunk 33]
```diff
@@ -440,7 +667,9 @@ export class Room extends EventEmitter implements Sayable {
   }
 
   /**
-   * @todo document me
+   * Get room's owner from the room.
+   * Not recommend, because cannot always get the owner
```
Comment (zixia) [inline src/room.ts]: please add `@deprecated` because it is very useless, almost no use.

File: src/wechaty.ts [hunk 34]
```diff
@@ -239,40 +246,151 @@ export class Wechaty extends EventEmitter implements Sayable {
...
+   * @returns {this}
+   * @example
+   * ```ts
+   * wechaty.on('room-leave', (room: Room, leaverList: Contact[]) => {
+   *   const nameList = leaverList.map(c => c.name()).join(',')
+   *   console.log(`Room ${room.topic()} lost member ${nameList}`)
+   * })
+   * ```
    */
   public on(event: 'room-leave' , listener: (this: Wechaty, room: Room, leaverList: Contact[]) => void): this
```
Comment (zixia) [inline src/wechaty.ts]: Please add a notice about the silent leave:

> If someone leaves the room by themselves, wechat will not notice other people in the room, so the bot will never get the "leave" event.

File: src/message.ts (hunk 24 above)
Comment (lijiarui) [inline src/message.ts]: How about the latest one?

Comment (kis87988) [inline src/message.ts]: it looks nice:)

File: docs/index.md [hunk 35]
```diff
@@ -65,190 +213,1040 @@ Wechaty.instance() // Singleton
...
+  console.log('You have already set an alias for contact ' + contact.name() + ':' + alias)
+}
+```
+**Example** *(SET the alias for a contact)*  
+```js
+const ret = await contact.alias('lijiarui')
+if (ret) {
+  console.log(`change ${contact.name()}'s alias successfully!`)
+} else {
+  console.error('failed to change ${contact.name()}'s alias!')
```
Comment (zixia) [inline docs/index.md]:
```diff
- console.error('failed to change ${contact.name()}'s alias!')
+ console.error(`failed to change ${contact.name()}`s alias!')
```

File: src/contact.ts [hunk 36]
```diff
@@ -1,7 +1,8 @@
 /**
+ *   @ignore
```
Comment (zixia) [inline src/contact.ts]: Could we move `@ignore` to the end of the comment, instead of at the beginning of?

File: src/message.ts [hunk 37]
```diff
@@ -410,44 +543,82 @@ export class Message implements Sayable {
...
+   *     console.log('Bot REPLY: hello world')
+   *     await m.say(new MediaMessage(__dirname + '/wechaty.png'))
+   *     console.log('Bot REPLY: Image')
+   *   }
+   * })
    */
-  public typeApp(): AppMsgType {
-    if (!this.rawObj) {
-      throw new Error('no rawObj')
+  public say(textOrMedia: string | MediaMessage, replyTo?: Contact|Contact[]): Promise<any> {
```
Comment (zixia) [inline src/message.ts]: Why did you rewrite so many original methods, could you get rid of all the unrelated changes?

Because it makes the diff shows lots of mess, and make the review becomes a hard job.

File: src/message.ts [hunk 38]
```diff
@@ -531,6 +701,59 @@ export class Message implements Sayable {
     return contactList
   }
 
+  /**
+   * Get the type from the message.
+   *
+   * @see {@link MsgType}
+   * @returns {MsgType}
+   */
+  public type(): MsgType {
```
Comment (zixia) [inline src/message.ts]: The same question as above: your diff should not include unrelated changes.

Comment (lijiarui) [inline src/message.ts]: ok

File: docs/index.md (hunk 35 above)
Comment (zixia) [inline docs/index.md]: ??

File: src/contact.ts [hunk 39]
```diff
@@ -150,25 +167,27 @@ export class Contact implements Sayable {
...
-   * Get the weixin number from a contact
+   * Get the weixin number from a contact.
+   *
    * Sometimes cannot get weixin number due to weixin security mechanism, not recommend.
-   * @returns {string | null}
    *
+   * @deprecated
+   * @returns {string | null}
    * @example
-   * ```ts
```
Comment (zixia) [inline src/contact.ts]: Could please confirm it for me that we do not need "\`\`\`ts" block inside `@example`?

File: src/contact.ts [hunk 40]
```diff
@@ -277,52 +278,40 @@ export class Contact implements Sayable {
   /**
    * Contact gender
    *
-   * @returns Gender.Male(2) | Gender.Female(1) | Gender.Unknown(0)
```
Comment (zixia) [inline src/contact.ts]: Just use `Gender.Male` `Gender.Female` and `Gender.Unknown`, there's no need to refer the number value of them because that's why ENUM type exists: Do not rely on the magic numbers.

So does the other ENUM types.

File: src/contact.ts [hunk 41]
```diff
@@ -447,27 +445,30 @@ export class Contact implements Sayable {
   }
 
   /**
-   * find contact by `name` or `alias`
+   * The way to search Contact
+   *
+   * @typedef    ContactQueryFilter
+   * @property   {string} name    -the name-string set by user-self, should be called name
```
Comment (zixia) [inline src/contact.ts]: `-the name` to `- The name`?

File: src/contact.ts [hunk 42]
```diff
@@ -718,20 +661,20 @@ export class Contact implements Sayable {
...
+   * @param {(string | MediaMessage)} textOrMedia
+   * @returns {Promise<boolean>}
    * @example
-   * ```ts
+   * const contact = await Contact.find({name: 'lijiarui'})         // change 'lijiarui' to any of your contact name in wechat
    * await contact.say('welcome to wechaty!')
-   * ```
+   * await contact.say(new MediaMessage(__dirname + '/wechaty.png') // put the filePath you want to send here
    */
-  public async say(text: string)
```
Comment (zixia) [inline src/contact.ts]: Why you deleted the overloading function declaration?

They should be kept because those are for TypeScript IDE Intelligence.

File: src/message.ts [hunk 43]
```diff
@@ -160,6 +160,29 @@ export interface MsgTypeMap {
   // , MessageTypeValue: MessageTypeName
 }
 
+/**
+ *
+ * Enum for AppMsgType values.
+ *
+ * @enum {number}
+ * @property {number} TEXT                    - 1      for TEXT
```
Comment (zixia) [inline src/message.ts]: We should just let users know AppMsgType.IMG, without the exact number.

When we use Wechaty, we should just use AppMsgType.IMG instead of `2`

File: src/message.ts [hunk 44]
```diff
@@ -396,12 +522,18 @@ export class Message implements Sayable {
     return null
   }
 
+  public content(): string
+
   /**
-   * @todo document me
+   * @private
    */
-  public content(): string
```
Comment (zixia) [inline src/message.ts]: Please keep the overloading method declaration.

File: src/room.ts [hunk 45]
```diff
@@ -170,24 +187,63 @@ export class Room extends EventEmitter implements Sayable {
...
+    * @property   {string}  topic - Get topic event, emitted when someone change room topic.
+    * @property   {string}  leave - Emit when anyone leave the room.<br>
+    *                               If someone leaves the room by themselves, wechat will not notice other people in the room, so the bot will never get the "leave" event.
+    */
+
   /**
-   * @todo document me
+   * @desc       Room Class Event Function
+   * @typedef    RoomEventFunction
+   * @property   {Function} room-join       -(this: Room, inviteeList: Contact[] , inviter: Contact)  => void
```
Comment (zixia) [inline src/room.ts]: Add a space? `- (this...`

File: src/contact.ts (hunk 39 above)
Comment (lijiarui) [inline src/contact.ts]: yes, I looked into the doc and confirmed, then changed all ``` inside @example

File: src/contact.ts (hunk 40 above)
Comment (lijiarui) [inline src/contact.ts]: well, m.type() return too much type, are you sure use like this?

File: src/message.ts (hunk 43 above)
Comment (lijiarui) [inline src/message.ts]: do you mean change `1` to `AppMsgType .TEXT` here?

Comment (zixia) [inline src/message.ts]: I think so, of there's no other problem?

File: src/contact.ts (hunk 40 above)
Comment (zixia) [inline src/contact.ts]: Yes. Just use the ENUM names please.

File: src/contact.ts (hunk 39 above)
Comment (zixia) [inline src/contact.ts]: Great, thanks.

File: src/message.ts (hunk 43 above)
Comment (lijiarui) [reply src/message.ts]: After I thought more if any want to know the number for more development, such as ipad protocol or xposed, maybe we should tell them the wechat raw number, So I add both `AppMsgType .TEXT`  and `1` here.

Comment (zixia) [reply src/message.ts]: I'll not argue with you about that, It's ok.

But I'd like to assume that the other protocols are different with the Web.

File: src/contact.ts [hunk 46]
```diff
@@ -1,7 +1,8 @@
 /**
+ *
  *   Wechaty - https://github.com/chatie/wechaty
  *
- *   Copyright 2016-2017 Huan LI <zixia@zixia.net>
+ *   @copyright 2016-2017 Huan LI <zixia@zixia.net>
```
Comment (hczhcz) [inline src/contact.ts]: It would be better if all files share the same header (copyright info, license, etc)

Comment (zixia) [reply src/contact.ts]: @hczhcz thanks! I'll do that later by an util script written by me at https://github.com/Chatie/wechaty/blob/master/script/update-license.ts, in case you are interested at.

**Comment by binsee** [comment]
look forward to merge

**Comment by lijiarui** [comment]
@binsee Could you help to approve this pull request? then @zixia can merge this, thanks!

**Comment by zixia** [comment]
Great! Thanks for all approving this PR, appreciate that!
